coreax/score_matching.py

```diff
@@ -545,7 +551,13 @@ def score_function(x_: Shaped[Array, " n d"]) -> Shaped[Array, " n d"]:
             :param x_: The :math:`n \times d` data vectors we wish to evaluate the score
                 function at
             """
-            # Check format
+            # Check format of input array. We use atleast_2d from JAX to perform
+            # conversions here. This is faster than using the custom function
+            # _atleast_2d_consistent in coreax.data, and whilst these two functions
```

Comment:
> This is faster than using the custom function `_atleast_2d_consistent`

I'm not sure this is true:

```
> python -m timeit --setup "import jax.numpy as jnp; from coreax.data import _atleast_2d_consistent; arr = jnp.arange(100)" -- "_atleast_2d_consistent(arr)"
50000 loops, best of 5: 5.14 usec per loop
> python -m timeit --setup "import jax.numpy as jnp; from coreax.data import _atleast_2d_consistent; arr = jnp.arange(100)" -- "jnp.atleast_2d(arr)"
50000 loops, best of 5: 5.67 usec per loop
```
so the two seem to have nearly identical running times.

The issue instead seems to be that because the output shape is different, we have to do more processing on lines 565 and 568 (as their output shape and output values become different.)